sklearn/utils/estimator_checks.py

```diff
@@ -403,6 +403,39 @@ def pairwise_estimator_convert_X(X, estimator, kernel=linear_kernel):
     return X
 
 
+def _generate_sparse_matrix(X_csr):
+    """ Generate matrices in multiple formats for CSR,CSC and COO matrices
+
+        Parameters
+        ----------
+
+        X_csr: CSR Matrix
+            Input matrix in CSR format
+
+        Returns
+        -------
+
+        out: iter(Matrices)
+            In format['dok', 'lil', 'dia', 'bsr', 'csr', 'csc', 'coo',
+             'coo_64', 'csc_64', 'csr_64']
+    """
+
+    for sparse_format in ['dok', 'lil', 'dia', 'bsr', 'csr', 'csc', 'coo']:
+        yield {"format": sparse_format, "X": X_csr.asformat(sparse_format)}
+
+    X_coo = X_csr.asformat('coo')
+    X_coo.row = X_coo.row.astype('int64')
+    X_coo.col = X_coo.col.astype('int64')
+
+    for sparse_format in ['csc', 'csr']:
```

Comment:
Make this conditional on having scipy >= 0.14 installed (that's the right version, yeah?)

Reply:
I meant to include the COO case above, also